### PR TITLE
Initial setup for CI

### PR DESCRIPTION
**Pull request by aditi-s3**
- Apply linting to Network, SlowDetector and CrashReports module. These have violations that need to be addressed
- Updated linting config 
- Caution: Indent Rule for swiftformat is disabled as File Headers with License violate this rule and need to be addressed
- build job in CI is commented as repo is not buildable yet, once repo builds this needs to be re-visited

### CODE REVIEW

**Comment by aditi-s3** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by SenNeonoveNoci — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,92 +1,20 @@
...
-          zip ${GITHUB_WORKSPACE}/SauceLabsTestApp.zip $(find ~/$BUILD_FOLDER/$BUILD_PATH/SauceLabsTestApp.app -type f)
-      - name: Upload test app to Sauce Labs
-        id: sl-upload
-        run: |
-          cd /Users/runner/work/splunk-otel-ios/splunk-otel-ios/
-          item_id=$(curl -u "${{secrets.SAUCELABS_USER}}:${{secrets.SAUCELABS_KEY}}" --location \
-            --request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
-            --form 'payload=@"SauceLabsTestApp.zip"' \
-            --form 'name="SauceLabsTestApp.zip"' \
-            --form 'description="'"$GITHUB_SHA"'"' | jq -r .item.id)
```
Comment (carlosmcevilly) [inline .github/workflows/ci.yml:63]: Do we know that jq will be installed on the build machine, or if not do we have some requirements document that lists it?

**Review by carlosmcevilly — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aditi-s3) [reply .github/workflows/ci.yml:63]: We don't know. The saucelabs tests are currently failing in the main branch, and we need to procure a team account and fix this pipeline. Hence, for now this is removed in feature/next-gen branch

Comment (carlosmcevilly) [reply .github/workflows/ci.yml:63]: And now I notice it was a removal, not an addition... thanks.

**Pull request merged**